plugins/source_metrics_test.go

```diff
@@ -0,0 +1,37 @@
+package plugins
+
+import "testing"
+
+func TestSourceMetrics(t *testing.T) {
+	s := &SourceMetrics{
+		TableClient: make(map[string]map[string]*TableClientMetrics),
+	}
+	s.TableClient["test_table"] = make(map[string]*TableClientMetrics)
+	s.TableClient["test_table"]["testExecutionClient"] = &TableClientMetrics{
+		Resources: 1,
+		Errors:    2,
+		Panics:    3,
+	}
+	if s.TotalResources() != 1 {
+		t.Fatal("expected 1 resource")
+	}
+	if s.TotalErrors() != 2 {
+		t.Fatal("expected 2 error")
+	}
+	if s.TotalPanics() != 3 {
+		t.Fatal("expected 3 panics")
+	}
+
+	other := &SourceMetrics{
+		TableClient: make(map[string]map[string]*TableClientMetrics),
+	}
+	other.TableClient["test_table"] = make(map[string]*TableClientMetrics)
+	other.TableClient["test_table"]["testExecutionClient"] = &TableClientMetrics{
+		Resources: 1,
+		Errors:    2,
+		Panics:    3,
+	}
+	if !s.Equal(other) {
```

Comment:
I don't this this test will catch if someone adds a field but forgets to add it to `Equal`

Reply:
This wont catch it but this should be caught in code-review. Also reflection can miss things and/or have incorrect results. I think there is no good way to ensure this apart from code-review but I would say this is reasonable for an equality function.